web/pandas/pdeps/0017-backwards-compatibility-and-deprecation-policy.md

```diff
@@ -0,0 +1,84 @@
+# PDEP-17: Backwards compatibility and deprecation policy
+
+- Created: 27 June 2024
+- Status: Under discussion
+- Discussion: [#59125](https://github.com/pandas-dev/pandas/issues/59125)
+- Author: [Abdulaziz Aloqeely](https://github.com/Aloqeely)
+- Revision: 1
+
+## Abstract
+
+This PDEP defines pandas' backwards compatibility and deprecation policy.
+
+## Motivation
+
+Having a clear backwards compatibility and deprecation policy is crucial to having a healthy ecosystem. We want to ensure users can rely on pandas being stable while still allowing the library to evolve.
+
+This policy will ensure that users have enough time to deal with deprecations while also minimizing disruptions on downstream packages' users.
+
+## Scope
+
+This PDEP covers pandas' approach to backwards compatibility and the deprecation and removal process.
+
+The decision making process for deprecating features is out of scope for this PDEP.
+
+## Background
+
+pandas uses a loose variant of semantic versioning.  
```

Comment:
Probably a misunderstanding on my end but what makes our releases a "loose variant" - I thought we followed it exactly?

Reply:
There are minor differences, the most notable is this rule

> Patch version Z (x.y.Z | x > 0) MUST be incremented if only backward compatible bug fixes are introduced.

Currently we allow for more than just bug fixes to be released in patch versions. (See v2.2.1 release notes)

Reply:
> Probably a misunderstanding on my end but what makes our releases a "loose variant" - I thought we followed it exactly?

My understanding of the "loose" aspect of it is that we are quite liberal in interpreting something as a breaking change or not. 

For example, our current policy (https://pandas.pydata.org/docs/dev/development/policies.html) has the note:

> pandas will sometimes make _behavior changing_ bug fixes, as part of minor or patch releases. Whether or not a change is a bug fix or an API-breaking change is a judgement call. 

If you strictly follow SemVer, that would require bumping the major version each time, I think.

We could also reference "Intended Effort Versioning" (EffVer) instead of / in addition to semantic versioning to explain our scheme: https://jacobtomlinson.dev/effver/, although we would have our own explanation of what changes are done in each type of release.

Reply:
Those two points cover my motivation for that "loose semver" phrasing. Exactly what makes a change a "new feature" or a "backwards incompatible" is sometimes fuzzy.